sw/device/silicon_creator/rom/e2e/rom_e2e_bootstrap_rma_test.c

```diff
@@ -0,0 +1,73 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static const char *reset_reason_to_string(dif_rstmgr_reset_info_t reset_info) {
+  switch (reset_info) {
+    case kDifRstmgrResetInfoPor:
+      return "Por";
+    case kDifRstmgrResetInfoLowPowerExit:
+      return "LowPowerExit";
+    case kDifRstmgrResetInfoSw:
+      return "Sw";
+    case kDifRstmgrResetInfoHwReq:
+      return "Hw";
+    case kDifRstmgrResetInfoSysRstCtrl:
+      return "SysRstCtrl";
+    case kDifRstmgrResetInfoWatchdog:
+      return "Watchdog";
+    case kDifRstmgrResetInfoPowerUnstable:
+      return "PowerUnstable";
+    case kDifRstmgrResetInfoEscalation:
+      return "Escalation";
+    case kDifRstmgrResetInfoNdm:
+      return "Ndm";
+  }
+}
+
+// This function dumps the reset reasons contained within `bitfield` to enable
+// human-readable regular expressions in the test harness. The reasons are
+// printed in a deterministic order, separated by commas.
+static void log_reset_info(dif_rstmgr_reset_info_bitfield_t bitfield) {
+  enum { kContentLen = 255 };
+  char buf[kContentLen + 1] = {0};
+  size_t pos = 0;
+
+  for (size_t i = 0; i < ARRAYSIZE(kDifRstMgrResetInfoAll); i++) {
+    const dif_rstmgr_reset_info_t reason = kDifRstMgrResetInfoAll[i];
+    if (!(bitfield & reason)) {
+      continue;
+    }
+    const size_t remaining = kContentLen - pos;
+    const char *sep = i > 0 ? "," : "";
+    const size_t n = base_snprintf(&buf[pos], remaining, "%s%s", sep,
+                                   reset_reason_to_string(reason));
+    CHECK(n <= kContentLen);
+    CHECK(pos < kContentLen - n);
+    pos += n;
+  }
+  CHECK(buf[kContentLen] == '\0');
+  LOG_INFO("Reset reasons: %s", buf);
+}
+
+bool test_main(void) {
+  dif_rstmgr_reset_info_bitfield_t bitfield = rstmgr_testutils_reason_get();
```

Comment:
ROM puts the reset reason in the retention RAM, I don't think we need to go through rstmgr_testutils for this:
```suggestion
  uint32_t reset_reasons = retention_sram_get()->reset_reasons;
```